lib/irb/init.rb

```diff
@@ -395,33 +395,41 @@ def IRB.parse_opts(argv: ::ARGV)
   # Run the config file
   def IRB.run_config
     if @CONF[:RC]
-      begin
-        file = rc_file
+      rc_files.each do |rc|
         # Because rc_file always returns `HOME/.irbrc` even if no rc file is present, we can't warn users about missing rc files.
         # Otherwise, it'd be very noisy.
-        load file if File.exist?(file)
+        load rc if File.exist?(rc)
       rescue StandardError, ScriptError => e
-        warn "Error loading RC file '#{file}':\n#{e.full_message(highlight: false)}"
+        warn "Error loading RC file '#{rc}':\n#{e.full_message(highlight: false)}"
       end
     end
   end
 
   IRBRC_EXT = "rc"
   def IRB.rc_file(ext = IRBRC_EXT)
+    warn "rc_file is deprecated, please use rc_files instead."
+    rc_files(ext).first
+  end
+
+  def IRB.rc_files(ext = IRBRC_EXT)
     if !@CONF[:RC_NAME_GENERATOR]
+      @CONF[:RC_NAME_GENERATOR] ||= []
+      existing_rc_files = []
```

Comment:
Since it's not actually a collection of files or file paths, but a collection of Procs, I think it should be called `existing_rc_file_generators` instead. Otherwise the `CONF[:RC_NAME_GENERATOR] = existing_rc_files` below looks a bit confusing.